commit 32a909dea83cc0bb6de07e53939c1650151ec1b3 (refs/changes/99/1899/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-05-04T16:19:21+02:00 (2 years, 5 months ago)
    
    TOOLS-1764 'sdcadm up SERVICE@VERSION' fails for services where the image name is different

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 8ac6c2e..4d1f268 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -3446,7 +3446,8 @@ SdcAdm.prototype.genUpdatePlan = function genUpdatePlan(options, cb) {
                     } else if (ch.version) {
                         self.updates.listImages({
                             version: ch.version,
-                            name: ch.service.name
+                            name:
+                                self.config.imgNameFromSvcName[ch.service.name]
                         }, function (iErr, img) {
                             if (iErr) {
                                 return nextChange(new errors.UpdateError(
